test/create-user.test.ts

```diff
@@ -0,0 +1,154 @@
+import "mocha";
+import "reflect-metadata";
+import { expect } from "chai";
+import { createHash } from "crypto";
+import { UserRepository } from "../src/data-source";
+import { getQuery, getMutation } from "./utils";
+import { getToken } from "../src/token";
+
+describe("create user test", () => {
+  //basic input object for testing
+  let input = {
+    profession: "teste",
+    password: "Teste123",
+    name: "teste1",
+    email: "teste@teste.com",
+    dateOfBirth: "01/01/2000",
+  };
+  let token = "";
+
+  //deleting users from repository and resetting basic input
+  beforeEach(async () => {
+    await UserRepository.clear();
+    input = {
+      profession: "teste",
+      password:
+        "ebdf496f67651cddf6aaa1f0b130f1b99ce9e2e93dc2503d926edcff15aee668",
+      name: "teste1",
+      email: "teste@teste.com",
+      dateOfBirth: "01/01/2000",
+    };
+    await UserRepository.save(input);
+
+    const existingUser = await UserRepository.findOneBy({
+      email: "teste@teste.com",
+    });
+    const testUser = {
+      profession: existingUser.profession,
+      name: existingUser.name,
+      dateOfBirth: existingUser.dateOfBirth,
+      email: existingUser.email,
+      id: existingUser.id,
+    };
+    token = getToken(testUser, true);
+
+    input = {
+      profession: "teste2",
+      password: "Teste123",
+      name: "teste2",
+      email: "teste2@teste.com",
+      dateOfBirth: "01/01/2000",
+    };
+  });
+
+  //setting up usersQuery
+  const queryBody = {
+    query: `query {users}`,
+    variables: {},
+  };
+
+  //setting up createUserMutation
+  const createMutation = (input) => {
```

Comment:
poderia tipar qual é o input
```suggestion
  const createMutation = (input: InputType) => {
```